libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2756,8 +2756,18 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			);
 
 		if (!funType->bound())
-			if (auto contractType = dynamic_cast<ContractType const*>(exprType))
-				requiredLookup = contractType->isSuper() ? VirtualLookup::Super : VirtualLookup::Virtual;
+		{
+			if (auto typeType = dynamic_cast<TypeType const*>(exprType))
+			{
+				if (
+					auto contractType = dynamic_cast<ContractType const*>(typeType->actualType());
+					contractType->isSuper()
+				)
+					requiredLookup = VirtualLookup::Super;
```

Comment:
Are you sure it always is a contract type?
```suggestion
				auto contractType = dynamic_cast<ContractType const*>(typeType->actualType());
				if (contractType && contractType->isSuper())
					requiredLookup = VirtualLookup::Super;
```

Reply:
Fixed.